src/goto-symex/frame.h

```diff
@@ -42,8 +42,37 @@ struct framet
   // loop and recursion unwinding
   struct loop_infot
   {
+    /// Incremental counter on how many branches this loop has had killed.
+    std::size_t children_too_complex = 0;
+
+    /// Set of loop ID's that have been blacklisted.
+    std::set<irep_idt> blacklisted_loops;
+
+    // Points to the parent loop if this is a nested one.
+    loop_infot *parent_loop = nullptr;
+
+    // Is this loop currently active.
+    bool is_active = false;
     unsigned count = 0;
     bool is_recursion = false;
+
+    /// Returns how many levels deep this loop is in a nested stack.
+    /// 0 = not nested, +1 per level.
+    int get_nesting_level()
+    {
+      if(nesting_level == -1)
+      {
+        if(parent_loop != nullptr)
+          nesting_level = parent_loop->get_nesting_level() + 1;
+        else
+          nesting_level = 0;
+      }
+
+      return nesting_level;
+    }
+
+  private:
+    int nesting_level = -1;
```

Comment:
This is a bit unclear, I guess -1 is used too mean this has not been computed yet. To make that clearer you could replace this field and the `get_nesting_level()` method by a field `lazyt<int> nesting_level = lazy([]{return parent_loop == nullptr ? 0 : parent_loop->nesting_level.force() + 1;})`